src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs

```diff
@@ -87,6 +101,16 @@ public static bool TryParse(string? value, [NotNullWhen(true)] out NuGetVersion?
                             metadata: buildMetadata ?? string.Empty,
                             originalVersion: originalVersion);
 
+                        lock (ParsedNuGetVersionsMapping)
+                        {
+                            if (ParsedNuGetVersionsMapping.Count >= ParsedNuGetVersionsMappingMaxEntries)
```

Comment:
Out of curiosity, what's the reason we clear the cache above the threshold?

Reply:
I didn't want unbounded growth. I saw this grow to a little over two hundred when opening Roslyn.sln, so I can't imagine many solutions will exceed this, but it felt safer to bound the growth just in case.

Reply:
Totally onboard with bounding the growth. Looks like we clear out the cache entirely though once we hit the bound. Am I missing something?

Reply:
Gotcha, I was thinking the cache was purged between sessions. The code I was imagining would just avoid adding the item to the cache if we're at the limit. Something like:

                            
    if (ParsedNuGetVersionsMapping.Count < ParsedNuGetVersionsMappingMaxEntries)
    {
        ParsedNuGetVersionsMapping[value] = version;
    }

Reply:
I think that it's best to just evacuate the cache if we hit the limit and let it start over. With the code you mention, I would have concerns about old and currently no longer used version information being in the cache over more relevant version information that was just requested.

Reply:
Totally fair. The usage pattern is really going to determine the "right" approach here. I don't know enough to make a call with any confidence.